shared/CommGroupDescription.F90

```diff
@@ -105,8 +115,19 @@ subroutine comm_group_range(this,my_node, my_rank, nodes_sizes, start_node, star
          endif
       endif
 
-      ! case 3, empty group
-      if (this%npes == 0 .and. this%nnodes ==0) then
+      ! case 3, npes per node
+      if (this%npes_per_node > 0) then
+         _ASSERT( all(nodes_sizes == nodes_sizes(1)) , " all nodes should have the same amount of cores")
+         next_rank  = start_rank + this%npes_per_node
+         next_node  = 0 ! no used
+         if (start_rank <=my_rank .and. my_rank < next_rank) then
+            _ASSERT( .not. IamInGroup, "I have been inluded in the other group")
```

Comment:
About all I can say about this is you misspelled `included`. The rest is...complex! :)

Reply:
fixed